src/main/java/co/novu/common/rest/RetryInterceptor.java

```diff
@@ -0,0 +1,62 @@
+package co.novu.common.rest;
+
+import java.io.IOException;
+
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class RetryInterceptor implements Interceptor{
+	
+	private final int maxRetries;
+	private final int minRetryDelayMillis;
+	private final int maxRetryDelayMillis;
+    private final int initialRetryDelayMillis;
+       
+    public RetryInterceptor(int maxRetries, int minRetryDelayMillis, int maxRetryDelayMillis, int initialRetryDelayMillis) {
+        this.maxRetries = maxRetries;
+        this.minRetryDelayMillis = minRetryDelayMillis;
+        this.maxRetryDelayMillis = maxRetryDelayMillis;
+        this.initialRetryDelayMillis = initialRetryDelayMillis;
+    }
+
+	@Override
+	public Response intercept(Chain chain) throws IOException {
+		Request request = chain.request();
+        Response response = null;
+        IOException lastException = null;
+
+        for (int retry = 0; retry < maxRetries; retry++) {
```

Comment:
We can use a while loop instead. Something like this `while (!response.isSuccessful() && retry < maxRetries) {}`
This will ensure that we're attempting the retrial only when the last call was not successful.

Also, note that the check for a successful call should be based on the the status codes specified [here](https://github.com/novuhq/novu/pull/4462#issuecomment-1757483022). You can have a utility function that does this check